test/common/upstream/health_check_fuzz.cc

```diff
@@ -221,6 +260,215 @@ void TcpHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type,
   }
 }
 
+void GrpcHealthCheckFuzz::allocGrpcHealthCheckerFromProto(
+    const envoy::config::core::v3::HealthCheck& config) {
+  health_checker_ = std::make_shared<TestGrpcHealthCheckerImpl>(
+      *cluster_, config, dispatcher_, runtime_, random_,
+      HealthCheckEventLoggerPtr(event_logger_storage_.release()));
+  ENVOY_LOG_MISC(trace, "Created Test Grpc Health Checker");
+}
+
+void GrpcHealthCheckFuzz::initialize(test::common::upstream::HealthCheckTestCase input) {
+  allocGrpcHealthCheckerFromProto(input.health_check_config());
+  cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+      makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectStreamCreate(0);
+  health_checker_->start();
+  ON_CALL(runtime_.snapshot_, getInteger("health_check.min_interval", _))
+      .WillByDefault(testing::Return(45000));
+
+  if (DurationUtil::durationToMilliseconds(input.health_check_config().initial_jitter()) != 0) {
+    test_sessions_[0]->interval_timer_->invokeCallback();
+  }
+  if (input.health_check_config().has_reuse_connection()) {
```

Comment:
you could remove this if block by using Envoy's protobuf get or default macro PROTOBUF_GET_WRAPPED_OR_DEFAULT

Reply:
also, will this affect the interval callback (which is called before?)

Reply:
Hey, this is used across all three? Should I change it to macro for the other two as well? Also, this reuse_connection boolean doesn't affect anything until a respond. Reuse connection being false means it blows away client after every response, which is why I persist this over time, affecting interval timer callback logic in respond methods.

Reply:
Changed to a macro call across all three health checkers.